### ISSUE TITLE
plotting peak centers in output. feature request/contrib

### ISSUE THREAD

**Issue by anomalosepia** [fooof-tools/fooof#145]
Hi, in the process of tuning fit parameters for some of my data I was having trouble identifying the locations of the fit peaks for individual fits. I modified the plotting in fm.py to add in vertical lines between the background and the peak fits. It's a little grubby, but it works and I find it very helpful when I'm trying to get an idea of how things are working. 

I think this would make a nice addition to the visualizations, particularly as you're getting started with using FOOOF. 

I just paste this at line 62 in fm.py

```
        #print(log_freqs)
        fitYforAmp = np.array([np.interp(x[0], fm.freqs, fm._ap_fit) for x in fm.peak_params_])
        peakYforAmp = np.array([x[1] for x in fm.peak_params_]) + fitYforAmp #

        #this is ugly but it works
        if log_freqs == False:
            for x, y1,y2 in zip(fm.peak_params_,fitYforAmp,peakYforAmp):
                plt.plot(x[0],y1,x[0],y2,  marker = 'o', linewidth  =3)
                plt.plot([x[0],x[0]],[y1,y2],  marker = 'o', linewidth  =3)
        if log_freqs:
            for x, y1,y2 in zip(fm.peak_params_,fitYforAmp,peakYforAmp):
                x = np.log10(x)
                plt.plot(x[0],y1,x[0],y2,  marker = 'o', linewidth  =3)
                plt.plot([x[0],x[0]],[y1,y2],  marker = 'o', linewidth  =3) 
```

**Comment by TomDonoghue** [comment]
Hey @anomalosepia - thanks for the suggestion! I've been playing with this idea a bit, and trying to figure out the best way to offer this - as I agree that some extra viz options for 'seeing' individual peaks would be useful. 

I've played with a couple options here, drawing either a line similar to what you did, added a shaded gaussian, or both. Do you have any suggestions, based on your use case, for what would be most useful to you?

![Screen Shot 2019-11-20 at 3 02 39 PM](https://user-images.githubusercontent.com/7727566/69286256-9d04c180-0ba7-11ea-8b0c-2053ac290989.png)

**Comment by anomalosepia** [comment]
I think that the bottom right is probably the best, it's busy, but it's also good to see the overlap of the distributions, having all 4 options is great though. Thank you so much. 

I will say that we had a hell of a time finding parameters and agreeing on them from both an empirical and theoretical perspective. It seems like the fits get less stable with messier (pediatric) data. I've really struggled when trying to run it in in a time-frequency kind of way in a 2s sliding window across a signal.

**Comment by anomalosepia** [comment]
Hey!
You saw our paper!

https://twitter.com/Tomdonoghue/status/1197958708262629376

**Comment by TomDonoghue** [comment]
Okay, for the plots, everything as above is now added to #152, with the addition of also using full lines across the whole plot, and combining different things to visualize plots. 

![Screen Shot 2020-01-19 at 7 58 29 PM](https://user-images.githubusercontent.com/7727566/72697794-1c563900-3af6-11ea-93b7-a3fe41d2b2a8.png)

@anomalosepia :
That's your paper! That's super cool! I hadn't connected the paper to you before, because I think I just saw your Github account name. 

Also, regarding:
> I will say that we had a hell of a time finding parameters and agreeing on them from both an 
> empirical and theoretical perspective. It seems like the fits get less stable with messier 
> (pediatric) data. I've really struggled when trying to run it in in a time-frequency kind of way in 
> a 2s sliding window across a signal.

Yeah, it can take a bit of tuning / trouble-shooting, especially with messy data. We are working on some ideas, including for time resolved analyses, but there's always limitations based on data clealiness. I would say right now FOOOF, current version isn't really built for or optimized for a sliding window, at least not that quickly - but hopefully we'll have something for that soon. Best I can suggest right now, is the more data and/or any tweaks that smooth out the power spectra will help. If you have any more comments / concerns / suggestions about any of this, feel free to open a new issue, or get in touch otherwise, Since this issue is mostly about the plots, I'll close it now.
